internal/controller/managedcluster_controller.go

```diff
@@ -761,6 +761,25 @@ func (r *ManagedClusterReconciler) reconcileCredentialPropagation(ctx context.Co
 				Reason:  hmc.SucceededReason,
 				Message: "vSphere CCM credentials created",
 			})
+		case "openstack":
+			l.Info("OpenStack creds propagation start")
+			if err := credspropagation.PropagateOpenStackSecrets(ctx, propnCfg, credential); err != nil {
```

Comment:
nit: the `".../credspropagation".PropagationCfg` symbol could be extended with the `credential`'s `Spec.IdentityRef` hence the signature of the `PropagateOpenStackSecrets` will be along with the rest of propagation functions. Up to the implementation and the code owning developer though

Reply:
Hmm, I am okay either ways. I guess we should do that (include identityRef into PropagationCfg struct) if it will be helpful for other providers in the future.
@eromanova thoughts? If you think its required, I can do that change in the same PR.

Reply:
I'd agree with Michael. Let's do it in this PR.

Reply:
Done.